### ISSUE TITLE
SIGSEGV in sampler::pick after client aborts a long prefill stream

### ISSUE THREAD

**Issue by aantidot** [Apolog1ze-Dev/QwFNfer#9]
Environment

- QwFNfer commit: f955dbf
- Linked against: GenerelSchwerz/llama.cpp tree (qwen4exp + lazy-mode API), CUDA 5080 16GB, 60GB RAM, Ubuntu 26.04
- Server: qwfn-server, console plan: ctx 262144, kv q4_0, RAM tier ~45.5GiB pinned, batch 2048, spec_block on, think xhigh (budget 6000)

Repro

1. Send a chat completion whose prompt is ~157k fresh tokens (no prefix cache hit).
2. During the (very long) prefill, abort the client connection (close the page / cancel fetch) before the first token.
3. Server log:  chatcmpl-...-4: client stopped reading after 156967 prompt + 1 generated tokens (402 s); stream dropped
4. Immediately after (or on the next request): SIGSEGV in the generation thread; engine dead, port stops answering.

Symbolized frames (addr2line -f -C on build/qwfn-server)

main::{lambda(chat_request&, server::prompt const&, stream_events&)#1}   <- crash thread
httplib DataSink response lambda (stream write path)
sampler::pick(float*, long)
sampler::top_indices(...)

i.e. the stream-write path continues sampling after the abort; logits buffer appears to be freed while sampler::pick/top_indices still read it.
Notes

- Short prompts / normal streams do not trigger this.
- Restarting the server recovers.
- No client-side timeout is involved — the abort happened while the client was still waiting for the first token.
